### PR TITLE
BITMAG-1133: Reference client understands salt input as a Hex String. This is NOT clear from the docs

### PR DESCRIPTION
**Pull request by Bohlski**
Fixed the [issue](https://sbforge.org/jira/browse/BITMAG-1133) and also changed underlying Base16Utils a bit. 

Base16Utils now enforces correct use - i.e. input string has even length and consists only of hexadecimal characters.

This entails that e.g. running the following using the commandline client:
`bitmag.sh put-file -f <file> -c <collection> -C <checksum> -R <checksum_type> -S salt`
will now throw an error, as the string _salt_ does not only consist of hexadecimals.

### CODE REVIEW

File: bitrepository-client/src/main/java/org/bitrepository/commandline/PutFileCmd.java [hunk 1]
```diff
@@ -108,16 +108,17 @@ protected void createOptionsForCmdArgumentHandler() {
...
         cmdHandler.addOption(checksumTypeOption);
+
         Option checksumSaltOption = new Option(Constants.REQUEST_CHECKSUM_SALT_ARG, Constants.HAS_ARGUMENT,
-                "[OPTIONAL] Used to request a salted checksum in the response. Requires the ChecksumType argument.");
+                Constants.REQUEST_CHECKSUM_SALT_DESC);
         checksumSaltOption.setRequired(ARGUMENT_IS_NOT_REQUIRED);
         cmdHandler.addOption(checksumSaltOption);
 
         Option deleteOption = new Option(Constants.DELETE_FILE_ARG, Constants.NO_ARGUMENT,
-                "If this argument is present, then the file will be removed from the server, when the chosen operation is complete.");
```
Comment (ole-v-v) [inline bitrepository-client/src/main/java/org/bitrepository/commandline/PutFileCmd.java:120]: Nitpicking now we're at it: remove the second comma, the one between "server" and "when".

File: bitrepository-core/src/main/java/org/bitrepository/common/utils/Base16Utils.java [hunk 2]
```diff
@@ -40,35 +42,22 @@ public static String decodeBase16(byte[] data) {
...
         ArgumentValidator.checkNotNullOrEmpty(hexString, "String hexString");
-        ArgumentValidator.checkTrue((hexString.length() % 2) == 0, "String hexString, '" + hexString
-                + "', must be an even number of characters.");
-
-        int len = hexString.length();
-        byte[] data = new byte[len / 2];
-        for (int i = 0; i < len; i += 2) {
-            data[i / 2] = (byte) ((Character.digit(hexString.charAt(i), 16) << 4)
-                    + Character.digit(hexString.charAt(i + 1), 16));
+        // TODO Java 17 has HexFormat.of().parseHex(s) - consider using instead
```
Comment (ole-v-v) [inline bitrepository-core/src/main/java/org/bitrepository/common/utils/Base16Utils.java:56]: I like this comment.

File: bitrepository-core/src/main/java/org/bitrepository/common/utils/Base16Utils.java [hunk 3]
```diff
@@ -40,35 +42,22 @@ public static String decodeBase16(byte[] data) {
...
-        StringBuilder sb = new StringBuilder(data.length * 2);
-        for (byte datum : data) {
-            int v = datum & 0xff;
-            if (v < 16) {
-                sb.append('0');
-            }
-            sb.append(Integer.toHexString(v));
-        }
-        return sb.toString();
+        return Hex.encodeHexString(data);
```
Comment (ole-v-v) [inline bitrepository-core/src/main/java/org/bitrepository/common/utils/Base16Utils.java:45]: Similarly to below one may consider adding a comment along the lines of `TODO Java 17 has HexFormat.of().formatHex(bytes); - consider using instead`.

**Review by ole-v-v — APPROVED** [review]
I like what I see.

File: bitrepository-core/src/main/java/org/bitrepository/common/utils/Base16Utils.java (hunk 3 above)
Comment (Bohlski) [reply bitrepository-core/src/main/java/org/bitrepository/common/utils/Base16Utils.java:45]: I actually did write this, but somehow I deleted it again. Will add. :)

**Review by m-atlantis — APPROVED** [review]

**Pull request merged**